### PR TITLE
[CI:DOCS] Add auth.json(5) link to login/logout docs

### PR DESCRIPTION
**Pull request by QiWang19**
Add auth.json(5) link to login/logout docs. Provide more details about the usage of auth.json by podman-login.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-login.1.md [hunk 1]
```diff
@@ -12,9 +12,12 @@ and password. If the registry is not specified, the first registry under [regist
...
 The username and password can also be set using the **username** and **password** flags.
 The path of the authentication file can be specified by the user by setting the **authfile**
-flag. The default path used is **${XDG\_RUNTIME\_DIR}/containers/auth.json**. If there is a valid
-username and password in the **authfile** , Podman will use those existing credentials if the user does not pass in a username.
-If those credentials are not present, Podman will then use any existing credentials found in **$HOME/.docker/config.json**.
+flag. The default path for reading and writing credentials is **${XDG\_RUNTIME\_DIR}/containers/auth.json**.
+Podman will use existing credentials if the user does not pass in a username.
+Podman will first search username and password in the **${XDG\_RUNTIME\_DIR}/containers/auth.json**, if they are not valid,
+Podman will then use any existing credentials found in **$HOME/.docker/config.json**.
+If those credentials are not present, Podman will create **${XDG\_RUNTIME\_DIR}/containers/auth.json** and store the username and password as base64 encoded string.
```
Comment (mheon) [inline docs/source/markdown/podman-login.1.md:19]: We request the password from the user here, right?

File: docs/source/markdown/podman-login.1.md [hunk 2]
```diff
@@ -12,9 +12,12 @@ and password. If the registry is not specified, the first registry under [regist
 from registries.conf will be used. **podman login** reads in the username and password from STDIN.
 The username and password can also be set using the **username** and **password** flags.
 The path of the authentication file can be specified by the user by setting the **authfile**
-flag. The default path used is **${XDG\_RUNTIME\_DIR}/containers/auth.json**. If there is a valid
-username and password in the **authfile** , Podman will use those existing credentials if the user does not pass in a username.
-If those credentials are not present, Podman will then use any existing credentials found in **$HOME/.docker/config.json**.
+flag. The default path for reading and writing credentials is **${XDG\_RUNTIME\_DIR}/containers/auth.json**.
+Podman will use existing credentials if the user does not pass in a username.
+Podman will first search username and password in the **${XDG\_RUNTIME\_DIR}/containers/auth.json**, if they are not valid,
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-login.1.md:17]:
```suggestion
Podman will first search for the username and password in the **${XDG\_RUNTIME\_DIR}/containers/auth.json**, if they are not valid,
```

File: docs/source/markdown/podman-login.1.md (hunk 1 above)
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-login.1.md:19]:
```suggestion
If those credentials are not present, Podman will create **${XDG\_RUNTIME\_DIR}/containers/auth.json** (if the file does not exist) and will then store the username and password as a base64 encoded string in it.
```

**Comment by TomSweeneyRedHat** [comment]
A couple of suggested changes, otherwise LGTM

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/podman/issues/6170

**Pull request merged**